src/services/gameleaderboard.service.ts

```diff
@@ -0,0 +1,280 @@
+    if (!userFieldData) {
+      return null;
+    }
+
+    const { name, value } = userFieldData;
+
+    const embed = new MessageEmbed();
+    embed.setTitle(`${this._gameEnumToString[game]} Leaderboard`);
+    embed.addField(name, value);
+    return embed;
+  }
+
+  private _createPlayerFieldData(user: User, entries: IUserOverallEntry[]): Maybe<EmbedFieldData> {
+    const [userEntry] = entries.filter((e) => e.player === user);
+    if (!userEntry) {
+      return null;
+    }
+
+    const userRank = entries.indexOf(userEntry);
+    return {
+      name: `${userRank + 1}. ${userEntry.player.username} (You)`,
+      value:
+        `*Wins:* ${userEntry.numWins}\n` +
+        `*Loses:* ${userEntry.numLoses}\n` +
+        `*Ties:* ${userEntry.numTies}`,
+    };
+  }
+
+  private async _parseCollectionData(
+    leaderboard: Collection<IGameLeaderBoardEntry>
+  ): Promise<IUserOverallEntry[]> {
+    return (await leaderboard.find().toArray())
+      .reduce((acc: IUserOverallEntry[], doc: IGameLeaderBoardEntry) => {
+        const stats = this._getOverallStats(doc);
+        if (stats) {
+          acc.push(stats);
+        }
+
+        return acc;
+      }, [])
+      .sort((a: IUserOverallEntry, b: IUserOverallEntry) => b.numWins - a.numWins);
+  }
+
+  public async createMatchupLeaderboardEmbed(userOne: User, userTwo: User, gameType: GameType) {
+    const leaderboard: Collection<IGameLeaderBoardEntry> = await this._gameEnumToCollection[
+      gameType
+    ]();
+    if (!leaderboard) {
+      this._loggerService.error(`Could not get leaderboard for ${gameType}`);
+      return 'Unable to get the leaderboards at this time';
+    }
+
+    const entries: IGameLeaderBoardEntry[] = await leaderboard.find().toArray();
+
+    const [userOneEntry] = entries.filter((e) => e.userId == userOne.id);
+
+    if (!userOneEntry) {
+      return `User \`${userOne.username}\` not found`;
+    }
+
+    const matchupGames = userOneEntry.games.filter((game: IGame) => game.opponent == userTwo.id);
+
+    let userOneWins = 0,
+      userTwoWins = 0,
+      ties = 0;
+
+    matchupGames.forEach((game: IGame) => {
+      switch (game.result) {
+        case GameResult.Won:
+          userOneWins++;
+          break;
+        case GameResult.Lost:
+          userTwoWins++;
+          break;
+        case GameResult.Tie:
+          ties++;
+          break;
+      }
+    });
+
+    const embed = new MessageEmbed();
+    embed.setTitle(`${this._gameEnumToString[gameType]} Matchup`);
+    embed.addField(
+      // `${userRank + 1}. ${userEntry.player.username} (You)`,
+      `${userOne.username} vs ${userTwo.username}`,
+      `__Wins:__ ${userOneWins} - ${userTwoWins}\n` +
+        `__Loses:__ ${userTwoWins} - ${userOneWins}\n` +
+        `__Ties:__ ${ties}`
+    );
+    return embed;
+  }
+
+  private _getOverallStats(entry: IGameLeaderBoardEntry): Maybe<IUserOverallEntry> {
+    const user = this._guildService.get().members.cache.get(entry.userId)?.user;
```

Comment:
what if cache miss? can we `||` this declaration with an actual fetch query to discord?

Reply:
also i thought we added some sort of util that has this fallback functionality?

Reply:
members.cache is the only way afaik